--- setup.py.orig	2015-04-02 18:20:13.000000000 -0700
+++ setup.py	2015-05-23 17:16:23.000000000 -0700
@@ -25,7 +25,7 @@
         install_requires=[
             'click >= 3.2',
             'Pygments >= 2.0',  # Pygments has to be Capitalcased. WTF?
-            'jedi == 0.8.1',    # Temporary fix for installation woes.
+            'jedi >= 0.9.0',
             'prompt_toolkit==0.26',
             'psycopg2 >= 2.5.4',
             'sqlparse >= 0.1.14'
