--- src/archives.c.orig	2015-09-16 05:50:15.000000000 -0700
+++ src/archives.c	2015-11-19 21:21:41.000000000 -0800
@@ -639,6 +639,7 @@
 {
   static struct varbuf conffderefn, symlinkfn;
   const char *usename;
+  char *s = NULL;
   struct filenamenode *usenode;
 
   struct conffile *conff;
@@ -707,6 +708,14 @@
 
   usenode = namenodetouse(nifd->namenode, tc->pkg, &tc->pkg->available);
   usename = usenode->name;
+  if (*usename == '.' && *usename + 1 == '/') {
+    usename += 1; /* Skip the leading `.' */
+  } else if (*usename != '/') {
+    s = malloc(strlen(usename) + 2); /* 1 for NULL, one for `/' we're going to add */
+    strcpy(s + 1, usename);
+    *s = '/';
+    usename = s;
+  }
 
   trig_file_activate(usenode, tc->pkg);
 
@@ -719,6 +728,9 @@
   }
 
   setupfnamevbs(usename);
+  if (s != NULL) {
+    free(s);
+  }
 
   statr= lstat(fnamevb.buf,&stab);
   if (statr) {
