--- scripts/Dpkg/Source/Archive.pm.orig	2015-04-09 14:40:47.000000000 -0700
+++ scripts/Dpkg/Source/Archive.pm	2015-04-30 03:12:46.000000000 -0700
@@ -48,7 +48,7 @@
     $spawn_opts{from_pipe} = \*$self->{tar_input};
     # Call tar creation process
     $spawn_opts{delete_env} = [ 'TAR_OPTIONS' ];
-    $spawn_opts{exec} = [ 'tar', '--null', '-T', '-', '--numeric-owner',
+    $spawn_opts{exec} = [ 'gnutar', '--null', '-T', '-', '--numeric-owner',
                             '--owner', '0', '--group', '0', '--format=gnu',
                             @{$opts{options}}, '-cf', '-' ];
     *$self->{pid} = spawn(%spawn_opts);
@@ -125,7 +125,7 @@
 
     # Call tar extraction process
     $spawn_opts{delete_env} = [ 'TAR_OPTIONS' ];
-    $spawn_opts{exec} = [ 'tar', '--no-same-owner', '--no-same-permissions',
+    $spawn_opts{exec} = [ 'gnutar', '--no-same-owner', '--no-same-permissions',
                             @{$opts{options}}, '-xf', '-' ];
     spawn(%spawn_opts);
     $self->close();
