--- src/archives.c.orig	2015-04-09 14:40:47.000000000 -0700
+++ src/archives.c	2015-04-30 03:04:24.000000000 -0700
@@ -772,6 +772,7 @@
 {
   static struct varbuf conffderefn, symlinkfn;
   const char *usename;
+  char *s = NULL;
   struct filenamenode *usenode;
 
   struct conffile *conff;
@@ -834,6 +835,14 @@
 
   usenode = namenodetouse(nifd->namenode, tc->pkg, &tc->pkg->available);
   usename = usenode->name;
+  if (*usename == '.' && *usename + 1 == '/') {
+    usename += 1; /* Skip the leading `.' */
+  } else if (*usename != '/') {
+    s = malloc(strlen(usename) + 2); /* 1 for NULL, one for `/' we're going to add */
+    strcpy(s + 1, usename);
+    *s = '/';
+    usename = s;
+  }
 
   trig_file_activate(usenode, tc->pkg);
 
@@ -847,6 +856,10 @@
 
   setupfnamevbs(usename);
 
+  if (s != NULL) {
+    free(s);
+  }
+
   statr= lstat(fnamevb.buf,&stab);
   if (statr) {
     /* The lstat failed. */
