--- lib/dpkg/tarfn.c.orig	2015-04-09 14:40:47.000000000 -0700
+++ lib/dpkg/tarfn.c	2015-04-30 02:48:06.000000000 -0700
@@ -40,6 +40,9 @@
 #define TAR_MAGIC_USTAR "ustar\0" "00"
 #define TAR_MAGIC_GNU   "ustar "  " \0"
 
+static const char ustarMagic[] = { 'u', 's', 't', 'a', 'r', '\0', '0', '0', '\0' };
+static const char gnutarMagic[] = { 'u', 's', 't', 'a', 'r', ' ', ' ', '\0' };
+
 struct tar_header {
 	char name[100];
 	char mode[8];
@@ -53,6 +56,7 @@
 
 	/* Only valid on ustar and gnu. */
 	char magic[8];
+	char MagicNumber[8]; /* POSIX: "ustar\000", GNU: "ustar  \0" (blank blank null) */
 	char user[32];
 	char group[32];
 	char devmajor[8];
@@ -60,6 +64,7 @@
 
 	/* Only valid on ustar. */
 	char prefix[155];
+	char Prefix[155]; /* POSIX ustar header */
 };
 
 /**
